### DISCUSSION TITLE
Speech Diarization inference scores (for pyannote.audio2)

### DISCUSSION THREAD

**Discussion by cembry26** [pyannote/pyannote-audio#643] [Q&A]
Hey @hbredin thank you for developing this awesome tool!

What are the three scores outputted per timestep on an inference call from the model?

example of how I'm getting at this:
```
model = PyanNet.load_from_checkpoint('/path/to/chpt.ckpt')

inference = Inference(model, device=torch.device('cuda'))

predictions = inference('/path/to/file.wav')
```

`predictions`

![dia_inference_output](https://user-images.githubusercontent.com/11167339/111368103-6fc50800-8663-11eb-9efa-7272df7b7860.png)

Where `predictions` now contains a sliding window feature with data of shape (736808, 3).

What do the three dimensions consist of? What colors do they correlate to in the notebook rendering of the sliding window?

Thank you so much for your hard work!

**Comment by hbredin** [comment]
Sorry for the delay in the answer: I was busy writing a paper describing the approach.
Please find it [here](https://arxiv.org/abs/2104.04045) - it should help you have a clearer idea of what is happening.
